### PR TITLE
Mob Faction refactor + Lizard wings

### PR DESCRIPTION
**Pull request by Cenrus**
https://github.com/tgstation/tgstation/pull/45828
https://github.com/tgstation/tgstation/pull/46037

Liz can now use flight potions for wings. Mob faction application has been improved slightly

image

## Why it's good for the game

Liz with wings

:cl: Nemvar, Wjohnston sprites and Mey Ha Zah for some help getting those sprites into a sane form
add: The flight potion now works with lizards. Use it as a liz to get cool derg wings.
code: Improved the way factions are applied to species.
fix: Shadowpeople no longer get attacked by faithless simplemobs.
/:cl:

### CODE REVIEW

**Comment by Cenrus** [comment]
Tested.

**Comment by ike709** [comment]
Conflicts.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Changing species while having wings keeps the effects of the wings, but removes the wing sprite, leading to the person just hovering.

**Comment by Cenrus** [comment]
Fixed issue with keeping wings.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1884,3 +1920,102 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 /datum/species/proc/start_wagging_tail(mob/living/carbon/human/H)
 
 /datum/species/proc/stop_wagging_tail(mob/living/carbon/human/H)
+
+///////////////
+//FLIGHT SHIT//
+///////////////
+
+/datum/species/proc/GiveSpeciesFlight(mob/living/carbon/human/H)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species.dm]: /datum/species/proc/give_species_flight(mob/living/carbon/human/H)

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1884,3 +1920,102 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+	flying_species = TRUE
+	if(isnull(fly))
+		fly = new
+		fly.Grant(H)
+	if(H.dna.features["wings"] != wings_icon)
+		mutant_bodyparts |= "wings"
+		H.dna.features["wings"] = wings_icon
+		H.update_body()
+
+/datum/species/proc/HandleFlight(mob/living/carbon/human/H)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species.dm]: /datum/species/proc/handle_flight(mob/living/carbon/human/H)

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -1884,3 +1920,102 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+	if(H.movement_type & FLYING)
+		if(!CanFly(H))
+			ToggleFlight(H)
+			return FALSE
+		return TRUE
+	else
+		return FALSE
+
+/datum/species/proc/CanFly(mob/living/carbon/human/H)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species.dm:1949]: /datum/species/proc/can_fly(mob/living/carbon/human/H)

File: code/modules/mob/living/carbon/human/species.dm [hunk 4]
```diff
@@ -1884,3 +1920,102 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+	H.stop_pulling()
+	if(buckled_obj)
+		buckled_obj.unbuckle_mob(H)
+		step(buckled_obj, olddir)
+	else
+		new /datum/forced_movement(H, get_ranged_target_turf(H, olddir, 4), 1, FALSE, CALLBACK(H, /mob/living/carbon/.proc/spin, 1, 1))
+	return TRUE
+
+//UNSAFE PROC, should only be called through the Activate or other sources that check for CanFly
+/datum/species/proc/ToggleFlight(mob/living/carbon/human/H)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species.dm]: /datum/species/proc/toggle_flight(mob/living/carbon/human/H)

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Decapitalise proc names

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**